src/benchmark/join_order_benchmark.cpp

```diff
@@ -13,97 +13,221 @@
+ * from the "third_party/join-order-benchmark" submodule. For an overview of the schema, see:
+ * https://doi.org/10.1007/s00778-017-0480-7 (Leis et al. "Query optimization through the looking glass, and what we
+ *                                                         found running the Join Order Benchmark")
  */
 
-using namespace hyrise;                // NOLINT
-using namespace std::string_literals;  // NOLINT
+using namespace hyrise;                // NOLINT(build/namespaces)
+using namespace std::string_literals;  // NOLINT(build/namespaces)
 
-/**
- * Each of the 21 JOB tables has one surrogate key. This function registers key constraints for all of them.
- */
 void add_key_constraints(std::unordered_map<std::string, BenchmarkTableInfo>& table_info_by_name) {
+  // Set all primary (PK) and foreign keys (FK) as defined in "Query optimization through the looking glass, and what we
+  // found running the Join Order Benchmark" (2.1 The IMDB data set, Fig. 2, p. 645).
+
+  // Get all tables.
   const auto& aka_name_table = table_info_by_name.at("aka_name").table;
+  const auto& aka_title_table = table_info_by_name.at("aka_title").table;
+  const auto& cast_info_table = table_info_by_name.at("cast_info").table;
+  const auto& char_name_table = table_info_by_name.at("char_name").table;
+  const auto& comp_cast_type_table = table_info_by_name.at("comp_cast_type").table;
+  const auto& company_name_table = table_info_by_name.at("company_name").table;
+  const auto& company_type_table = table_info_by_name.at("company_type").table;
+  const auto& complete_cast_table = table_info_by_name.at("complete_cast").table;
+  const auto& info_type_table = table_info_by_name.at("info_type").table;
+  const auto& keyword_table = table_info_by_name.at("keyword").table;
+  const auto& kind_type_table = table_info_by_name.at("kind_type").table;
+  const auto& link_type_table = table_info_by_name.at("link_type").table;
+  const auto& movie_companies_table = table_info_by_name.at("movie_companies").table;
+  const auto& movie_info_table = table_info_by_name.at("movie_info").table;
+  const auto& movie_info_idx_table = table_info_by_name.at("movie_info_idx").table;
+  const auto& movie_keyword_table = table_info_by_name.at("movie_keyword").table;
+  const auto& movie_link_table = table_info_by_name.at("movie_link").table;
+  const auto& name_table = table_info_by_name.at("name").table;
+  const auto& person_info_table = table_info_by_name.at("person_info").table;
+  const auto& role_type_table = table_info_by_name.at("role_type").table;
+  const auto& title_table = table_info_by_name.at("title").table;
+
+  // Set constraints.
+
+  // aka_name - 1 PK, 1 FK.
   aka_name_table->add_soft_key_constraint({{aka_name_table->column_id_by_name("id")}, KeyConstraintType::PRIMARY_KEY});
+  aka_name_table->add_soft_foreign_key_constraint({{aka_name_table->column_id_by_name("person_id")},
```

Comment:
Why even passing the type? If we have an enum class for the constraint types, the derived classes can set this on their own. Gonna add that.

Reply:
we could also provide methods like in `expression_functional.hpp`. So instead of writing:
```c++
table->add_soft_constraint(TableKeyConstraint{{table->column_id_by_name("id")}, 
                                              KeyConstraintType::PRIMARY_KEY});
table->add_soft_constraint(TableKeyConstraint{{table->column_id_by_name("name")}, 
                                              KeyConstraintType::UNIQUE});
table->add_soft_constraint(ForeignKeyConstraint{{table->column_id_by_name("nation")},
                                                table,  
                                                {nation_table->column_id_by_name("id")},
                                                nation_table});
table->add_soft_constraint(TableOrderConstraint{{table->column_id_by_name("id")},
                                                {table->column_id_by_name("name")}});
```

we could have:
```c++
table->add_soft_constraint(primary_key_({table->column_id_by_name("id")}));
table->add_soft_constraint(unique_({table->column_id_by_name("name")}));
table->add_soft_constraint(foreign_key_({table->column_id_by_name("nation")},
                                        nation_table, {nation_table->column_id_by_name("id")}));
table->add_soft_constraint(order_({table->column_id_by_name("id")},
                                  {table->column_id_by_name("name")}));
```